app/views/posts/index.html.erb

```diff
@@ -1 +1,40 @@
-<h1>User 1 all posts</h1>
+<div class="container">
+  <!-- User Information Section -->
+  <div class="user_info">
+    <div class="img">
+      <!-- Display user photo using Rails image_tag helper -->
+      <%= image_tag @user.photo, alt: "User Photo", class: "user-photo" %>
+    </div>
+    <div class="user">
+      <!-- Display user name with a link to user profile -->
+      <h2><%= link_to @user.name, user_path(@user) %></h2>
+      <!-- Display the number of posts by the user -->
+      <p><strong>Number of Posts:</strong> <%= @user.posts.count %></p>
+    </div>
+  </div>
+
+  <!-- Display the two most recent posts by the user -->
+  <% @posts.limit(2).each do |post| %>
+    <div class="post">
+      <h4>Post</h4>
+      <!-- Display the text content of the post -->
+      <p><%= post.text %></p>
+      <!-- Display the count of comments and likes for the post -->
+      <p class="count">Comments: <%= post.comments.count %> | Likes: <%= post.likes.count %></p>
+    </div>
+
+    <!-- Display the five most recent comments for each post -->
+    <div class="comments">
+      <% post.five_most_recent_comments.each do |comment| %>
+        <div class="comment">
+          <!-- Display the author's name with a link to their profile and the comment text -->
+          <p><%= link_to comment.author.name, user_path(comment.author) %>: <%= comment.text %></p>
```

Comment:
- [ ]  Please make sure your posts index page shows without errors after fixing the first error, for now, this error `undefined method `author'` is shown after fixing the first error it means the `author_id` Foreign is not  inside your comment table it's correct name is `user_id` as foreign key in you comment table so change `comment.author.name` and  `user_path(comment.author)` with `comment.user.name` and  `user_path(comment.user)` :+1: check below the screenshot of the error :slightly_smiling_face:

![Screenshot from 2023-12-15 22-21-38](https://github.com/MohammadYaser/blog-app/assets/45226848/b2d783ad-efa2-426b-8f1b-fbe46c5927bc)

```suggestion
          <p><%= link_to comment.user.name, user_path(comment.user) %>: <%= comment.text %></p>
```